src/relativeUrlPrefixRule.ts

```diff
@@ -0,0 +1,66 @@
+import { IOptions, IRuleMetadata, RuleFailure, Rules } from 'tslint/lib';
+import { SourceFile } from 'typescript/lib/typescript';
+import { NgWalker } from './angular/ngWalker';
+import * as ts from 'typescript';
+
+export class Rule extends Rules.AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: "The ./ prefix is standard syntax for relative URLs; don't depend on Angular's current ability to do without that prefix.",
+    descriptionDetails: 'See more at https://angular.io/guide/styleguide#style-05-04.',
+    rationale: 'A component relative URL requires no change when you move the component files, as long as the files stay together.',
+    ruleName: 'relative-url-prefix',
+    options: null,
+    optionsDescription: 'Not configurable.',
+    type: 'maintainability',
+    typescriptOnly: true
+  };
+
+  static readonly FAILURE_STRING = 'The ./ prefix is standard syntax for relative URLs. (https://angular.io/guide/styleguide#style-05-04)';
+
+  apply(sourceFile: SourceFile): RuleFailure[] {
+    return this.applyWithWalker(new RelativePathExternalResourcesRuleWalker(sourceFile, this.getOptions()));
+  }
+}
+
+export class RelativePathExternalResourcesRuleWalker extends NgWalker {
+  constructor(sourceFile: SourceFile, options: IOptions) {
+    super(sourceFile, options);
+  }
+
+  visitClassDecorator(decorator: ts.Decorator) {
```

Comment:
This method can be simplified if you use `visitNgComponent`.

Reply:
Using `visitNgComponent` was my first thought. Unfortunately, I can't access to 'templateUrl' et 'styleUrls'. It seams more efficient with inline template.